### PR TITLE
ServerTravel cleaning: Fix race condition and removing log. 

### PR DESCRIPTION
**Pull request by MatthewSandfordImprobable**
**Contributions**: We are not currently taking public contributions - see our [contributions](CONTRIBUTING.md) policy. However, we are accepting issues and we do want your [feedback](../README.md#give-us-feedback).

#### Description
There was a race where the server had loaded the map, the GSM had been created but the server did not have authority over the SpatialConstants::STARTUP_ACTOR_MANAGER_COMPONENT_ID component on the GSM. This lead to a failed check() in SetCanBeginPlay.

Also adding the !GlobalStateManager->IsReadyToCallBeginPlay() check.
This is just in case we load the map after we gain authority over the component on the GSM. When we gain autority the GSM has will have already called SetCanBeginPlay(true) so we shouldn't do this again.

Also removing the log as I don't think it is helpful. The flow of not doing the GlobalStateManager->SetCanBeginPlay(true) logic is perfectly valid. Essentially we only want to go down this path on server travel.

#### Tests
ServerTravel still functional. Log removed. 

STRONGLY SUGGESTED: How can this be verified by QA?
Will ask the reporters of the bug as this is a race. 

#### Primary reviewers
@joshuahuburn

### CODE REVIEW

File: SpatialGDK/Source/SpatialGDK/Private/EngineClasses/SpatialNetDriver.cpp [hunk 1]
```diff
@@ -618,18 +618,16 @@ void USpatialNetDriver::OnMapLoaded(UWorld* LoadedWorld)
...
 	if (IsServer())
 	{
-		if (GlobalStateManager != nullptr)
+		if (GlobalStateManager != nullptr &&
+			!GlobalStateManager->IsReadyToCallBeginPlay() &&
+			StaticComponentView->HasAuthority(GlobalStateManager->GlobalStateManagerEntityId, SpatialConstants::STARTUP_ACTOR_MANAGER_COMPONENT_ID))
 		{
-			// Increment the session id, so users don't rejoin the old game.
+			// ServerTravel - Increment the session id, so users don't rejoin the old game.
 			GlobalStateManager->SetCanBeginPlay(true);
```
Comment (m-samiec) [inline SpatialGDK/Source/SpatialGDK/Private/EngineClasses/SpatialNetDriver.cpp]: If GSM isn't ready when the map is loaded, what's the expected behaviour here? When will this get re-called?

Comment (MatthewSandfordImprobable) [reply SpatialGDK/Source/SpatialGDK/Private/EngineClasses/SpatialNetDriver.cpp]: GlobalStateManager::AuthorityChanged. When we gain authority over the STARTUP_ACTOR_MANAGER_COMPONENT_ID we call GlobalStateManager:: SetCanBeginPlay(true)

Comment (m-samiec) [reply SpatialGDK/Source/SpatialGDK/Private/EngineClasses/SpatialNetDriver.cpp]: What about the other three calls to the GSM?

Comment (MatthewSandfordImprobable) [reply SpatialGDK/Source/SpatialGDK/Private/EngineClasses/SpatialNetDriver.cpp]: HandleStartupOpQueueing will call TriggerBeginPlay.
SetAcceptingPlayers is set to true in GlobalStateManager::AuthorityChanged when we gain authority over DEPLOYMENT_MAP_COMPONENT_ID
We don't want to call IncrementSessionID on first session

File: SpatialGDK/Source/SpatialGDK/Private/EngineClasses/SpatialNetDriver.cpp [hunk 2]
```diff
@@ -618,18 +618,16 @@ void USpatialNetDriver::OnMapLoaded(UWorld* LoadedWorld)
 
 	if (IsServer())
 	{
-		if (GlobalStateManager != nullptr)
+		if (GlobalStateManager != nullptr &&
+			!GlobalStateManager->IsReadyToCallBeginPlay() &&
+			StaticComponentView->HasAuthority(GlobalStateManager->GlobalStateManagerEntityId, SpatialConstants::STARTUP_ACTOR_MANAGER_COMPONENT_ID))
```
Comment (joshuahuburn) [inline SpatialGDK/Source/SpatialGDK/Private/EngineClasses/SpatialNetDriver.cpp]: Can't we use `bool UGlobalStateManager::HasAuthority()` here?

File: SpatialGDK/Source/SpatialGDK/Private/EngineClasses/SpatialNetDriver.cpp [hunk 3]
```diff
@@ -618,18 +618,16 @@ void USpatialNetDriver::OnMapLoaded(UWorld* LoadedWorld)
 
 	if (IsServer())
 	{
-		if (GlobalStateManager != nullptr)
+		if (GlobalStateManager != nullptr &&
+			!GlobalStateManager->IsReadyToCallBeginPlay() &&
```
Comment (joshuahuburn) [inline SpatialGDK/Source/SpatialGDK/Private/EngineClasses/SpatialNetDriver.cpp]: Isn't it a little odd that we check `!GlobalStateManager->IsReadyToCallBeginPlay()` and then straight after call `GlobalStateManager->SetCanBeginPlay(true);` and `GlobalStateManager->TriggerBeginPlay();`

Comment (MatthewSandfordImprobable) [reply SpatialGDK/Source/SpatialGDK/Private/EngineClasses/SpatialNetDriver.cpp]: Well to be honest I wanted the check to be: !bCanBeginPlay
bCanBeginPlay is private on GSM with an accessor IsReadyToCallBeginPlay.

File: SpatialGDK/Source/SpatialGDK/Private/EngineClasses/SpatialNetDriver.cpp (hunk 2 above)
Comment (MatthewSandfordImprobable) [reply SpatialGDK/Source/SpatialGDK/Private/EngineClasses/SpatialNetDriver.cpp]: Is that guaranteed to work the same? HasAuthority does:
StaticComponentView->HasAuthority(GlobalStateManagerEntityId, SpatialConstants::SINGLETON_MANAGER_COMPONENT_ID)

They are checking different components so...

**Comment by MatthewSandfordImprobable** [comment]
Couple of comments left on comments.

File: SpatialGDK/Source/SpatialGDK/Private/EngineClasses/SpatialNetDriver.cpp (hunk 3 above)
Comment (joshuahuburn) [reply SpatialGDK/Source/SpatialGDK/Private/EngineClasses/SpatialNetDriver.cpp]: Maybe I'm missing the deeper meaning here xD 
!CanBeginPlay implies you then call BeginPlay?

Comment (MatthewSandfordImprobable) [reply SpatialGDK/Source/SpatialGDK/Private/EngineClasses/SpatialNetDriver.cpp]: If CanBeginPlay is false it means it seems valid to set it true with SetCanBeginPlay(true)

**Comment by MatthewSandfordImprobable** [comment]
Updated.

**Pull request merged**